pyvista/plotting/widgets.py

```diff
@@ -2623,6 +2625,202 @@ def _the_callback(widget, _event):
+
+        >>> import pyvista as pv
+        >>> p = pv.Plotter()
+        >>> def set_bg(color):
+        ...     def wrapped_callback():
+        ...         p.background_color = color
+        ...
+        ...     return wrapped_callback
+        ...
+        >>> p.add_radio_button_widget(
+        ...     set_bg('white'),
+        ...     'bgcolor',
+        ...     position=(10.0, 200.0),
+        ...     title='White',
+        ...     value=True,
+        ... )
+        >>> p.add_radio_button_widget(
+        ...     set_bg('lightblue'),
+        ...     'bgcolor',
+        ...     position=(10.0, 140.0),
+        ...     title='Light Blue',
+        ... )
+        >>> p.add_radio_button_widget(
+        ...     set_bg('pink'),
+        ...     'bgcolor',
+        ...     position=(10.0, 80.0),
+        ...     title='Pink',
+        ... )
+        >>> p.show()
+
+        """
+        if self.iren is None:  # pragma: no cover
+            raise RuntimeError('Cannot add a widget to a closed plotter.')
+
+        if radio_button_group not in self.radio_button_widget_dict:
+            self.radio_button_widget_dict[radio_button_group] = []
+        if title is not None:
+            if radio_button_group not in self.radio_button_title_dict:
+                self.radio_button_title_dict[radio_button_group] = []
+            button_title = self.add_text(
+                title, position=(position[0] + size + 10.0, position[1] + 7.5), font_size=15
+            )
+            self.radio_button_title_dict[radio_button_group].append(button_title)
+
+        color_on = Color(color_on)
+        color_off = Color(color_off)
+        background_color = Color(background_color, default_color=self.background_color)
+
+        def create_radio_button(fg_color, bg_color, size=size, smooth=2):  # numpydoc ignore=GL08
+            fg_color = np.array(fg_color.int_rgb)
+            bg_color = np.array(bg_color.int_rgb)
+
+            n_points = size**2
+            button = pyvista.ImageData(dimensions=(size, size, 1))
+            arr = np.array([bg_color] * n_points).reshape(size, size, 3)  # fill background
+
+            centre = size / 2
+            rad_outer = centre
+            rad_inner = centre - border_size
+            # Paint radio button with simple anti-aliasing
+            for i in range(size):
+                for j in range(size):
+                    distance = np.sqrt((i - size / 2) ** 2 + (j - size / 2) ** 2)
+                    if distance < rad_inner:
+                        arr[i, j] = fg_color
+                    elif rad_inner <= distance <= rad_inner + smooth:
+                        blend = (distance - rad_inner) / smooth
+                        arr[i, j] = (1 - blend) * fg_color + blend * bg_color
+                    elif rad_outer - 2 * smooth <= distance <= rad_outer:
+                        blend = abs(distance - rad_outer + smooth) / smooth
+                        arr[i, j] = (1 - blend) * fg_color + blend * bg_color
```

Comment:
```suggestion
            for i, j in product(range(size), range(size)):
                distance = np.sqrt((i - size / 2) ** 2 + (j - size / 2) ** 2)
                if distance < rad_inner:
                    arr[i, j] = fg_color
                elif rad_inner <= distance <= rad_inner + smooth:
                    blend = (distance - rad_inner) / smooth
                    arr[i, j] = (1 - blend) * fg_color + blend * bg_color
                elif rad_outer - 2 * smooth <= distance <= rad_outer:
                    blend = abs(distance - rad_outer + smooth) / smooth
                    arr[i, j] = (1 - blend) * fg_color + blend * bg_color
```

Reply:
@nathmay You can [incorporate feedback in your pull request using GitHub feature](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) :)

Reply:
@tkoyama010 Thank you! I realized this after already making additional commits. Can you tell I'm new at this? 😀